revolution_pi/gpio_pin.go

```diff
@@ -145,56 +244,61 @@ func (pin *gpioPin) PWMFreq(ctx context.Context, extra map[string]interface{}) (
 	if !pin.initialized {
 		return 0, errors.New("pin not initialized")
 	}
+	if !pin.isOutputPWM() && !pin.isOutputWord() {
+		return 0, fmt.Errorf("cannot get PWM Frequency, Pin %s is not a PWM pin", pin.Name)
+	}
 
 	b := make([]byte, 1)
-	n, err := pin.ControlChip.fileHandle.ReadAt(b, 112)
+	// all PWM pins use the same PWM frequency
+	n, err := pin.ControlChip.fileHandle.ReadAt(b, int64(pin.inputOffset+outputPWMFrequencyOffset))
 	if err != nil {
 		return 0, err
 	}
 	if n != 1 {
 		return 0, errors.New("unable to read PWM Frequency")
 	}
-	pin.ControlChip.logger.Infof("Current frequency: %#v", b)
+	pin.ControlChip.logger.Debugf("Current frequency step size: %#d", b)
 
-	return 0, nil
+	return stepSizeToFreq(b), nil
 }
 
-// SetPWMFreq sets the given pin to the given PWM frequency. For Raspberry Pis,
-// 0 will use a default PWM frequency of 800.
```

Comment:
maybe a more detailed comment explaining that the byte read from the register represents one of these 5 frequencies?